agents_infra/agents_infra/utils/communication.py

```diff
@@ -0,0 +1,32 @@
+from logtools import maybe_log_message
+from token_manager import TokenManager
+
+
+class AgentCommunicator(object):
+    def __init__(self, server_name, post_data_fn, logger):
+        self.server_name = server_name
+        self.post_data = post_data_fn
```

Comment:
The attribute name `post_data` is very similar to the method `post_data_with_auth`, which could be confusing. Consider renaming the attribute (e.g., `post_data_fn` or `_post_data_fn`) for clarity.